### PR TITLE
make ecdsa linux-compatible

### PR DESCRIPTION
**Pull request by omsmith**

### CODE REVIEW

File: src/D2L.Security.OAuth2/Keys/Default/EcDsaJsonWebKey.cs [hunk 1]
```diff
@@ -75,25 +78,16 @@ string y
 			};
 		}
 
-		private ECDsaCng BuildEcDsaCng() {
-			byte[] publicBlob = ECCPublicKeyBlobFormatter.Instance.BuildECCPublicBlob( this );
-			using( var cng = CngKey.Import( publicBlob, CngKeyBlobFormat.EccPublicBlob ) ) {
```
Comment (omsmith) [inline src/D2L.Security.OAuth2/Keys/Default/EcDsaJsonWebKey.cs:80]: Windows-only

File: src/D2L.Security.OAuth2/Keys/Default/EcDsaPrivateKeyProvider.cs [hunk 2]
```diff
@@ -7,36 +7,24 @@ namespace D2L.Security.OAuth2.Keys.Default {
...
-			using( var cngKey = CngKey.Create( m_algorithm, null, creationParams ) ) {
-				using( ECDsaCng ecDsa = new ECDsaCng( cngKey ) ) {
-					privateBlob = ecDsa.Key.Export( CngKeyBlobFormat.EccPrivateBlob );
-				}
-			}
+			var ecdsa = ECDsa.Create( m_curve );
+			var parameters = ecdsa.ExportParameters( includePrivateParameters: true );
 
 			D2LSecurityToken result = m_d2lSecurityTokenFactory.Create( () => {
-				using( var cng = CngKey.Import( privateBlob, CngKeyBlobFormat.EccPrivateBlob ) ) {
```
Comment (omsmith) [inline src/D2L.Security.OAuth2/Keys/Default/EcDsaPrivateKeyProvider.cs:34]: Windows-only

**Review by j3parker — APPROVED** [review]

File: src/D2L.Security.OAuth2/Keys/Default/D2LSecurityToken.cs [hunk 3]
```diff
@@ -55,16 +55,12 @@ internal sealed partial class D2LSecurityToken : SecurityKey {
...
-						return SecurityAlgorithms.EcdsaSha512;
-					}
-
-					throw new NotSupportedException();
+					return ecDsaSecurityKey.KeySize switch {
+						256 => SecurityAlgorithms.EcdsaSha256,
+						384 => SecurityAlgorithms.EcdsaSha384,
+						521 => SecurityAlgorithms.EcdsaSha512,
+						_ => throw new NotSupportedException(),
+					};
```
Comment (omsmith) [inline src/D2L.Security.OAuth2/Keys/Default/D2LSecurityToken.cs:63]: Don't love this part, but, easiest, and correct afaik given other constraints.

**Comment by omsmith** [comment]
Before:

```
Test run for /home/omsmith/dev/d2l/D2L.Security.OAuth2/test/D2L.Security.OAuth2.IntegrationTests/bin/Release/net50/D2L.Security.OAuth2.IntegrationTests.dll (.NETCoreApp,Version=v5.0)
Microsoft (R) Test Execution Command Line Tool Version 16.9.4
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...
A total of 1 test files matched the specified pattern.
  Failed ValidateAsync_BadSignature_Fails [34 ms]
  Error Message:
   OneTimeSetUp: System.AggregateException : One or more errors occurred. (Unknown curve: ECDSA_P521)
  ----> System.Exception : Unknown curve: ECDSA_P521

  Failed ValidateAsync_GoodSignature_Succeeds [34 ms]
  Error Message:
   OneTimeSetUp: System.AggregateException : One or more errors occurred. (Unknown curve: ECDSA_P521)
  ----> System.Exception : Unknown curve: ECDSA_P521

  Failed WebCrypto ES256 [34 ms]
  Error Message:
   OneTimeSetUp: System.AggregateException : One or more errors occurred. (Unknown curve: ECDSA_P521)
  ----> System.Exception : Unknown curve: ECDSA_P521

  Failed WebCrypto ES384 [34 ms]
  Error Message:
   OneTimeSetUp: System.AggregateException : One or more errors occurred. (Unknown curve: ECDSA_P521)
  ----> System.Exception : Unknown curve: ECDSA_P521

  Failed WebCrypto ES512 [34 ms]
  Error Message:
   OneTimeSetUp: System.AggregateException : One or more errors occurred. (Unknown curve: ECDSA_P521)
  ----> System.Exception : Unknown curve: ECDSA_P521

  Failed ValidateAsync_KeyIdNotInAuthService_Fails [34 ms]
  Error Message:
   OneTimeSetUp: System.AggregateException : One or more errors occurred. (Unknown curve: ECDSA_P521)
  ----> System.Exception : Unknown curve: ECDSA_P521

Setup failed for test fixture D2L.Security.OAuth2.Validation.AccessTokens.AccessTokenValidatorTests+EcDsa
System.AggregateException : One or more errors occurred. (Unknown curve: ECDSA_P521)
  ----> System.Exception : Unknown curve: ECDSA_P521

Failed!  - Failed:     6, Passed:    23, Skipped:     0, Total:    29, Duration: 2 s - /home/omsmith/dev/d2l/D2L.Security.OAuth2/test/D2L.Security.OAuth2.IntegrationTests/bin/Release/net50/D2L.Security.OAuth2.IntegrationTests.dll (net5.0)
```

**Comment by omsmith** [comment]
After

```
Test run for /home/omsmith/dev/d2l/D2L.Security.OAuth2/test/D2L.Security.OAuth2.IntegrationTests/bin/Release/net472/D2L.Security.OAuth2.IntegrationTests.dll (.NETFramework,Version=v4.7.2)
Microsoft (R) Test Execution Command Line Tool Version 16.9.4
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...
A total of 1 test files matched the specified pattern.
System.IO.FileNotFoundException: Could not find 'mono' host. Make sure that 'mono' is installed on the machine and is available in PATH environment variable.
   at Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Helpers.DotnetHostHelper.GetMonoPath()
   at Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Hosting.DefaultTestHostManager.GetTestHostProcessStartInfo(IEnumerable`1 sources, IDictionary`2 environmentVariables, TestRunnerConnectionInfo connectionInfo)
   at Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Client.ProxyOperationManager.SetupChannel(IEnumerable`1 sources, String runSettings)
   at Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Client.ProxyExecutionManager.StartTestRun(TestRunCriteria testRunCriteria, ITestRunEventsHandler eventHandler)

Test Run Aborted.
Test run for /home/omsmith/dev/d2l/D2L.Security.OAuth2/test/D2L.Security.OAuth2.IntegrationTests/bin/Release/net50/D2L.Security.OAuth2.IntegrationTests.dll (.NETCoreApp,Version=v5.0)
Microsoft (R) Test Execution Command Line Tool Version 16.9.4
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...
A total of 1 test files matched the specified pattern.

Passed!  - Failed:     0, Passed:    29, Skipped:     0, Total:    29, Duration: 2 s - /home/omsmith/dev/d2l/D2L.Security.OAuth2/test/D2L.Security.OAuth2.IntegrationTests/bin/Release/net50/D2L.Security.OAuth2.IntegrationTests.dll (net5.0)
```

**Review by j3parker — APPROVED** [review]

**Pull request merged**